File: frontend/src/types/connection/SessionLeftMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

/**
 * Tells the server that this client left the transfer screen, so other devices
 * stop seeing it as busy. Sent after a close by the peer, where the screen
 * outlives the connection. Carries no payload.
 */
export class SessionLeftMessage implements ITypedMessage {
    public readonly type = MessageType.SESSION_LEFT;
    public msg: Record<string, never> = {};
}

File: backend/backend/src/ConnectionComponent/Common/Api/DTOs/SessionLeftMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.ConnectionComponent.Common.Api.DTOs;

/// <summary>
/// Reports that the sending client left the transfer screen after the peer
/// closed the connection. The client counts as available again from here on.
/// Carries no payload.
/// </summary>
public class SessionLeftMessage : ITypedMessage
{
    public static string TypeString => "session-left";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;
}
